src/main/java/randoop/sequence/VariableRenamer.java

```diff
@@ -61,49 +69,103 @@ static String getVariableName(Type type) {
       if (type.equals(JavaTypes.BYTE_TYPE)) {
         return "byte";
       }
-      // otherwise, use the first character of the type name
+      if (type.equals(JavaTypes.INT_TYPE)) {
+        return "int";
+      }
+      if (type.equals(JavaTypes.BOOLEAN_TYPE)) {
+        return "bool";
+      }
+      if (type.equals(JavaTypes.FLOAT_TYPE)) {
+        return "float";
+      }
+      if (type.equals(JavaTypes.DOUBLE_TYPE)) {
+        return "double";
+      }
+      if (type.equals(JavaTypes.SHORT_TYPE)) {
+        return "short";
+      }
+
+      // Otherwise, use the first character of the type name.
       return type.getName().substring(0, 1);
     }
 
-    // special cases:  Object, String, Class
+    // Special cases:  Object, String, Class.
     if (type.isObject()) {
       return "obj";
     } else if (type.isString()) {
       return "str";
     } else if (type.equals(JavaTypes.CLASS_TYPE)) {
       return "cls";
     }
-    // TODO: add more special cases:
-    //  * if it implements List, use "list"
-    //  * if it implements Iterator, use "itor"
-    // ... more?
 
     if (type.isParameterized()) {
       ClassOrInterfaceType classType = (ClassOrInterfaceType) type;
       String varName = classType.getSimpleName().toLowerCase();
       if (varName.equals("class")) {
         return "cls";
       }
-      for (TypeArgument argument : classType.getTypeArguments()) {
+
+      // Special cases for parameterized types.
+      if (varName.contains("iterator")) {
+        // Iterator takes precedence, in cases like ListIterator.
+        varName = "itor";
+      } else if (varName.contains("list")) {
+        // List comes before array, in cases like ArrayList.
+        varName = "list";
+      } else if (varName.contains("set")) {
```

Comment:
This seems too broad to me; the letters `set` (in any capitalization) might appear in a type name.
I suggest that the rules be case-sensitive, based on the `getSimpleName()`; the fallthough case can downcase that.